### DISCUSSION TITLE
Type of empty array

### DISCUSSION THREAD

**Discussion by fleble** [scikit-hep/awkward#2186] [Q&A] [answered]
Dear experts,

I am writing ROOT files using uproot. One problem I face is when a branch is empty, thus having unknown type, which cannot be written down with uproot.

```bash
>>> ak.__version__
1.5.1
>>> uproot.__version__
4.3.7
```

This script
```python
ak_array = ak.Array([[], [], []])
ak_array = ak.values_astype(ak_array, np.float64)

tree = {"branch": ak_array}

with uproot.recreate("test.root") as file:
    file["test"] = tree
```
 terminates with the following error:
 ```
 TypeError: cannot write Awkward Array type to ROOT file:

    var * unknown
 ```

And:
 ```bash
>>> ak_array = ak.Array([[], [], []])
>>> ak.values_astype(ak_array, np.float64)
 <Array [[], [], []] type='3 * var * unknown'> 
 ```
 
 Is it possible to give a type to empty to empty awkward array?

**Comment by fleble** [comment]
After some attempts, I found a trick by concatenating an ak array of the desired type, and filtering out the part that was concatenated.

**Comment by jpivarski** [comment, marked as answer]
Converted into an issue: https://github.com/scikit-hep/uproot5/issues/822.

And also #2188.
